### PR TITLE
[CBRD-21807] Fix retrieving records with desc property and NULL fields

### PR DESCRIPTION
**Pull request by lucianlazar2211**
http://jira.cubrid.org/browse/CBRD-21807

This will implement proper usage of a `MIN_VALUE` column type in a `MIDXKEY`. It will also do the correct comparison of two midxkeys, since the last implementation would fail to correctly handle the case when a `NULL` value will be compared against a respective value from the other `MIDXKEY`, which had its corresponding column set to `MIN_VALUE`.

### CODE REVIEW

File: src/object/object_primitive.c [hunk 1]
```diff
@@ -8566,7 +8566,10 @@ pr_midxkey_compare (DB_MIDXKEY * mul1, DB_MIDXKEY * mul2, int do_coercion, int t
 		    }
 		  else
 		    {
-		      c = DB_GT;
```
Comment (ribram) [inline src/object/object_primitive.c]: Fix coding style:
1. ```c = mul1->min_max_val.type == MIN_COLUM ? DB_LT : DB_GT;```
or
2. Add brackets {} to if/else scopes.

Also, shouldn't something similar be added to ```else if (mul2->min_max_val.position == i)```?

File: src/storage/btree.c [hunk 2]
```diff
@@ -14804,14 +14804,14 @@ btree_coerce_key (DB_VALUE * keyp, int keysize, TP_DOMAIN * btree_domainp, int k
...
-			  ;	/* nop */
+			  minmax = BTREE_COERCE_KEY_WITH_MAX_VALUE;
 			}
 		    }
 		}
 
 	      if (minmax == BTREE_COERCE_KEY_WITH_MIN_VALUE)
 		{
-		  if (dsize < keysize)
+		  if (dsize <= keysize)
```
Comment (ribram) [inline src/storage/btree.c]: What if dsize > keysize?

File: src/object/object_primitive.c (hunk 1 above)
Comment (eseokoh) [reply src/object/object_primitive.c]: I'd like to propose to use `if-else` to clearly reveal the covered and uncovered branch.
Conditional statement is simple but uncertain from the point of line coverage.
I think we have to make all the branches covered. 
please also change line 8525 and 8537.

File: src/storage/btree.c (hunk 2 above)
Comment (eseokoh) [reply src/storage/btree.c]: I don't think there's such case: (dsize > keysize).
Do you think it is possible?

I'm not sure dsize==keysize case also need to set min_max_val.
If the case also needs, do you think we need `if`?

Comment (ribram) [reply src/storage/btree.c]: dsize == keysize needs min_max_val. ```if``` should be removed.

Comment (eseokoh) [reply src/storage/btree.c]: yes, indeed :)

**Pull request merged**